fastapi_opa/auth/auth_saml.py

```diff
@@ -29,15 +40,65 @@ async def authenticate(
         auth = await self.init_saml_auth(request_args)
 
         if "acs" in request.query_params:
+            print(datetime.utcnow(), '--acs--')
             return await self.assertion_consumer_service(auth, request_args)
         # potentially extend with logout here
+        elif 'sso' in request.query_params:
+            print(datetime.utcnow(), '--sso--')
+            return await self.single_sign_on(auth)
+            # TODO: check below code
+            # If AuthNRequest ID need to be stored in order to later validate it, do instead
+            # sso_built_url = auth.login()
+            # request.session['AuthNRequestID'] = auth.get_last_request_id()
+            # return redirect(sso_built_url)
+        elif 'sso2' in request.query_params:
```

Comment:
This should be tested.

Reply:
Done

Reply:
The add attrs are outside of the sso function, will think about how to test this.